### PR TITLE
fix(docs): Update the skore.train_test_split example

### LINKED ISSUE #782: fix(docs): Update the `skore.train_test_split` example
Following new PRs:
- https://github.com/probabl-ai/skore/pull/780
- https://github.com/probabl-ai/skore/pull/781
- https://github.com/probabl-ai/skore/pull/783

See https://github.com/probabl-ai/skore/issues/684

### PR DESCRIPTION
**Pull request by sylvaincom**
Fix #782

### CODE REVIEW

File: examples/plot_05_train_test_split.py [hunk 1]
```diff
@@ -138,39 +143,48 @@
 X_train, X_test, y_train, y_test = skore.train_test_split(
```
Comment (augustebaum) [inline examples/plot_05_train_test_split.py:143]: Line 140 "eventual" is not a good translation of "eventuellement" (false friend). Maybe "potential"?

File: examples/plot_05_train_test_split.py [hunk 2]
```diff
@@ -138,39 +143,48 @@
...
-print(X_train, "\n", y_train, "\n", X_test, "\n", y_test)
+X_train_explicit = X_train.copy()
 
 # %%
 # Moreover, when passing ``X`` and ``y`` explicitly, the ``X``'s are always returned
 # before the ``y``'s, even when they are inverted:
 
 # %%
-arr = np.arange(10).reshape((5, 2))
+arr = X.copy()
```
Comment (augustebaum) [inline examples/plot_05_train_test_split.py:153]: Line 149 this could be in a "Note" block

File: examples/plot_05_train_test_split.py [hunk 3]
```diff
@@ -138,39 +143,48 @@
...
 arr_train, arr_test, X_train, X_test, y_train, y_test = skore.train_test_split(
     arr, y=y, X=X, test_size=0.2, random_state=0
 )
-print(X_train, "\n", y_train, "\n", X_test, "\n", y_test)
+X_train_explicit_inverted = X_train.copy()
+
+print("When expliciting, with the small typo, are the `X_train`'s still the same?")
+print(np.allclose(X_train_explicit, X_train_explicit_inverted))
```
Comment (augustebaum) [inline examples/plot_05_train_test_split.py:160]: Nice, this is a full test!

File: examples/plot_05_train_test_split.py [hunk 4]
```diff
@@ -138,39 +143,48 @@
...
+
 
 # %%
 # Automatic diagnostics: raising methodological warnings
 # ------------------------------------------------------
 #
+# In this section, we show how skore can provide methodological checks.
+#
+# Class-imbalance
```
Comment (augustebaum) [inline examples/plot_05_train_test_split.py:169]:
```suggestion
# Class imbalance
```

Comment (augustebaum) [reply examples/plot_05_train_test_split.py:169]: Plus all other instances of "class-imbalance"

File: examples/plot_05_train_test_split.py [hunk 5]
```diff
@@ -138,39 +143,48 @@
...
-X = [[1]] * 4
-y = [0, 1, 1, 1]
+X = np.arange(10_000).reshape((5_000, 2))
+y = [0] * 4_000 + [1] * 1_000
 
 # %%
-# In that case, :func:`skore.train_test_split` raises a warning telling the user that
-# there is class imbalance:
+# In that case, :func:`skore.train_test_split` raises a ``HighClassImbalanceWarning``
+# warning telling the user that there is class imbalance:
```
Comment (augustebaum) [inline examples/plot_05_train_test_split.py:187]:
```suggestion
# telling the user that there is class imbalance:
```

Comment (augustebaum) [reply examples/plot_05_train_test_split.py:187]: not sure which is better, i'll let you decide

Comment (augustebaum) [reply examples/plot_05_train_test_split.py:187]: same for all other instances of "... raise a `MyWarning` warning"

File: examples/plot_05_train_test_split.py (hunk 1 above)
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:143]: Done

File: examples/plot_05_train_test_split.py (hunk 4 above)
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:169]: Done

File: examples/plot_05_train_test_split.py [hunk 6]
```diff
@@ -58,8 +58,9 @@
 # %%
 import numpy as np
 
-X, y = np.arange(10).reshape((5, 2)), range(5)
-print(X, "\n", y)
+X = np.arange(10).reshape((5, 2))
+y = np.arange(5)
+print(f"{X}\n{y}")
```
[the remark this answers is not in the document]
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:63]: Done

File: examples/plot_05_train_test_split.py [hunk 7]
```diff
@@ -70,7 +71,7 @@
 X_train, X_test, y_train, y_test = sklearn_train_test_split(
     X, y, test_size=0.2, random_state=0
 )
-print(X_train, "\n", y_train, "\n", X_test, "\n", y_test)
+print(f"{X_train}\n{y_train}\n{X_test}\n{y_test}")
```
[the remark this answers is not in the document]
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:74]: Favoring f-strings
https://github.com/probabl-ai/skore/pull/784#discussion_r1851698944

File: examples/plot_05_train_test_split.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:63]: Favoring f-strings
https://github.com/probabl-ai/skore/pull/784#discussion_r1851698944

File: examples/plot_05_train_test_split.py (hunk 2 above)
Comment (augustebaum) [reply examples/plot_05_train_test_split.py:153]: Oh, I thought you could put the code cells inside the block, no worries

File: examples/plot_05_train_test_split.py (hunk 5 above)
Comment (augustebaum) [reply examples/plot_05_train_test_split.py]: I mean that this passage isn't the only place where the docs say "warning" twice like that, so if you change it here then you should change it in the other places too

Comment (sylvaincom) [reply examples/plot_05_train_test_split.py]: Going with "warning"

File: examples/plot_05_train_test_split.py [hunk 8]
```diff
@@ -132,52 +137,86 @@
...
 
 # %%
-# Hence, skore recommends the users to take into account this class-imbalance, that
+# Hence, skore recommends the users to take into account this class imbalance, that
 # they might have missed, in their modelling strategy.
-# skore provides methodological checks.
+
+# %%
+# Moreover, skore also detects class imbalance with a class that has too few samples
+# with a ``HighClassImbalanceTooFewExamplesWarning`` warning:
```
[the remark this answers is not in the document]
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py:200]: Oh, ok

[the remark this answers is not in the document]
Comment (sylvaincom) [reply examples/plot_05_train_test_split.py]: Done

**Review by augustebaum — APPROVED** [review]

File: examples/plot_05_train_test_split.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (MarieS-WiMLDS) [reply examples/plot_05_train_test_split.py]: I didn't know this way of doing things, nice!

**Review by MarieS-WiMLDS — APPROVED** [review]
good for me :) !

**Comment by sylvaincom** [comment]
Thanks all!

**Pull request merged**